### PR TITLE
UX improvements

### PR DESCRIPTION
**Pull request by prophetl33t**
Tool name/args spell check (for example if user enters extract-npc (s missing), datatool suggests corrected variant if available).
Add ToolAtribute HideFromHelp which hides tool from help list (mostly used for debug tools).
Merge DebugDumpSTU and ExtractDebugType into one tool and rename it to extract-stu-type.
Remove DebugVoodoo, DumpFlags(was same as ListFlags) and RenderStateScript (superseeded by extract-stu-type) and move commented out tools to Unused folder.
Some minor descriptions/names changes.

### CODE REVIEW

File: DataTool/ToolLogic/Extract/ExtractFlags.cs [hunk 1]
```diff
@@ -97,6 +97,9 @@ public class ExtractFlags : ICLIFlags {
         // [CLIFlag(Default = false, Flag = "convert-bnk", Help = "Convert .bnk files to .wem", Parser = new[] { "DataTool.Flag.Converter", "CLIFlagBoolean" })]
         // public bool ConvertBnk;
 
+        [CLIFlag(Default = false, Flag = "XML", Help = "Convert STUs to xml when extracted with ExtractDebugType", Parser = new[] { "DataTool.Flag.Converter", "CLIFlagBoolean" })]
```
Comment (yretenai) [inline DataTool/ToolLogic/Extract/ExtractFlags.cs:100]: Flags should be lowercase.

File: DataTool/ToolAttribute.cs [hunk 2]
```diff
@@ -8,6 +8,7 @@ public class ToolAttribute : Attribute {
         public bool IsSensitive { get; set; } = false;
         public Type CustomFlags { get; set; } = null;
         public string Name { get; set; } = string.Empty;
+        public bool HideFromHelp { get; set; } = false;
```
Comment (yretenai) [inline DataTool/ToolAttribute.cs:11]: why not IsSensitive?

Comment (prophetl33t) [reply DataTool/ToolAttribute.cs]: fixd

File: DataTool/ToolLogic/Extract/Debug/Unused/ExtractDebugVCCE.cs [hunk 3]
```diff
@@ -7,7 +7,7 @@
 using static DataTool.Helper.IO;
 
 namespace DataTool.ToolLogic.Extract.Debug {
-    [Tool("extract-debug-vcce", Description = "Extract VCCE as png (debug)", CustomFlags = typeof(ExtractFlags), IsSensitive = true)]
+    [Tool("extract-debug-vcce", Description = "Extract VCCE as png (debug)",, CustomFlags = typeof(ExtractFlags), IsSensitive = true)]
```
Comment (ZingBallyhoo) [inline DataTool/ToolLogic/Extract/Debug/Unused/ExtractDebugVCCE.cs:10]: ah yes

File: DataTool/ToolLogic/Extract/Debug/Unused/ExtractDebugPackageFinder.cs [hunk 4]
```diff
@@ -6,7 +6,7 @@
 using static DataTool.Program;
 
 namespace DataTool.ToolLogic.Extract.Debug {
-    [Tool("extract-debug-packagefinder", Description = "Test package loading (debug)", CustomFlags = typeof(ExtractFlags), IsSensitive = true)]
+    [Tool("extract-debug-packagefinder", Description = "Test package loading (debug)",, CustomFlags = typeof(ExtractFlags), IsSensitive = true)]
```
Comment (ZingBallyhoo) [inline DataTool/ToolLogic/Extract/Debug/Unused/ExtractDebugPackageFinder.cs:9]: ah yes

File: DataTool/ToolLogic/Dump/DumpFlags.cs [hunk 5]
```diff
@@ -1,5 +0,0 @@
-﻿using DataTool.ToolLogic.List;
-
-namespace DataTool.ToolLogic.Dump {
-    public class DumpFlags : ListFlags { }
```
Comment (ZingBallyhoo) [inline DataTool/ToolLogic/Dump/DumpFlags.cs:4]: this existed to make it a seperate section in the help

File: DataTool/ToolLogic/Extract/Debug/ExtractDebugType.cs [hunk 6]
```diff
@@ -1,35 +0,0 @@
-﻿using System;
-using System.IO;
-using DataTool.Flag;
-using static DataTool.Program;
-using static DataTool.Helper.IO;
-
-namespace DataTool.ToolLogic.Extract.Debug {
-    [Tool("extract-debug-type", Description = "Extract type (debug)", CustomFlags = typeof(ExtractFlags), IsSensitive = true)]
-    public class ExtractDebugType : ITool {
```
Comment (ZingBallyhoo) [inline DataTool/ToolLogic/Extract/Debug/ExtractDebugType.cs:9]: this is not the same as dumping stus... why remove :(

File: DataTool/ToolLogic/Dump/DumpFlags.cs (hunk 5 above)
Comment (prophetl33t) [reply DataTool/ToolLogic/Dump/DumpFlags.cs:4]: 5Head

File: DataTool/ToolLogic/Extract/Debug/Unused/ExtractDebugPackageFinder.cs (hunk 4 above)
Comment (prophetl33t) [reply DataTool/ToolLogic/Extract/Debug/Unused/ExtractDebugPackageFinder.cs:9]: come on, these tools broken anyways if you uncomment them

Comment (Js41637) [reply DataTool/ToolLogic/Extract/Debug/Unused/ExtractDebugPackageFinder.cs:9]: Well they are now.

File: DataTool/ToolLogic/Extract/ExtractFlags.cs (hunk 1 above)
Comment (ZingBallyhoo) [reply DataTool/ToolLogic/Extract/ExtractFlags.cs]: nobody has ever wanted to see this on the help for normal extract modes
zingy

**Pull request merged**